### PR TITLE
Short names

### LINKED ISSUE #304: Use `_` in crate names
We are using hyphens in package names, but underscores pretty much everywhere else:

- File names
- Crate names
- Module names
- Function names

I don't understand why we don't use underscores in package names, too.  There are lots of Rust packages that use underscores, like serde_json.  This would eliminate the oddness that you put "icu-locale" in Cargo.toml, but then when you use it in code, you need to write "icu_locale".

### PR DESCRIPTION
**Pull request by zbraniecki**
Fixes #304.
Depends on #311

### CODE REVIEW

File: components/cldr_json_data_provider/Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "icu-cldr-json-data-provider"
+name = "icu_cldr_json_data_provider"
```
Comment (sffc) [inline components/cldr_json_data_provider/Cargo.toml:2]: Hmm.  `"icu_data_cldr"`?

File: components/data_provider/Cargo.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "icu-data-provider"
+name = "icu_data_provider"
```
Comment (sffc) [inline components/data_provider/Cargo.toml:2]: Maybe just `"icu_data"`, and then the class is called DataProvider?

**Review by sffc — COMMENTED** [review]
The package names for the data provider crates are long.  We should consider taking this opportunity to shorten them as well.  My suggestion:

- icu_testdata (status quo)
- icu_data (replaces icu_data_provider)
- icu_data_cldr (replaces icu_cldr_json_data_provider)
- icu_data_fs (replaces (icu_fs_data_provider)

**Comment by zbraniecki** [comment]
> The package names for the data provider crates are long. We should consider taking this opportunity to shorten them as well. My suggestion:
> 
>     * icu_testdata (status quo)
> 
>     * icu_data (replaces icu_data_provider)
> 
>     * icu_data_cldr (replaces icu_cldr_json_data_provider)
> 
>     * icu_data_fs (replaces (icu_fs_data_provider)

I'd rather go for `icu_provider`, `icu_cldr_provider`, `icu_fs_provider`.

**Comment by filmil** [comment]
Is there a way we could have the renames happen after 0.1 is tagged?
Unless you think renames are critical -- as they well may be, since after
0.1 sweeping renames would probably be harder.
I will say that I'm not exactly looking forward to merging that with my
upcoming changes, but I understand if it is necessary. :)

**Comment by sffc** [comment]
icu_provider works for me.

If we want to reduce disruptions to merging, we could start by renaming the packages and modules but keeping the file system paths the same.

**Comment by sffc** [comment]
@zbraniecki says he'll revisit this PR next week, so @filmil you are unblocked.

**Comment by zbraniecki** [comment]
I'm going to recreate a new PR based on the outcome of #304.

**Pull request closed without merging**